DataConnectors/Digital shadows/DigitalShadowsConnectorAzureFunction/DS_poller.py

```diff
@@ -41,10 +48,21 @@ def get_data(self):
         """
         triage_id = []
         try:
-            event_dataJSON = self.DS_obj.get_triage_events(str(self.before_time), str(self.after_time))
-            event_data = json.loads(event_dataJSON)
+            if(isinstance(self.event, int)):
+                event_dataJSON = self.DS_obj.get_triage_events_by_num(self.event)
+                event_data = json.loads(event_dataJSON)
+                self.date.post_event(self.event + int(len(event_data[:20])))
+            else:
+                event_dataJSON = self.DS_obj.get_triage_events(str(self.before_time), str(self.after_time))
+                event_data = json.loads(event_dataJSON)
+                event_num = int(event_data[0]['event-num'])
+                self.date.post_event(event_num + int(len(event_data[:20])))
+                logging.info("First poll from event number " + str(event_num))
+
+            
             logging.info("total number of events are " + str(len(event_data)))
-            for event in event_data:
+            
+            for event in event_data[:20]:
                 if(event is not None):
                     triage_id.append(event['triage-item-id'])
 
```

Comment:
Dont' we need to catch all exceptions at the end, so that we can log the exception details.

Reply:
I can change from specific exceptions to all exceptions, for the outermost try and except. But I want the function to fail at the unknown errors so that we will know there has been some error which has not yet been seen.